### ISSUE TITLE
Osmdroid App Stalling (only Suggestion)

### ISSUE THREAD

**Issue by InI4** [osmdroid/osmdroid#1309]
## Issue Type

[x] Bug (Minor)
[x] Documentation

## Description and/or steps/code to reproduce the problem

I've rarely started the Osmdroid app and am no expert with it. However today I gave it a look today and stalled the app several times. As it turned out, the solution was to do "View the intro again" up to the point, where new permissions get granted. Then the problems disappeared. 

For me, this is fine, it is probably good test (with and without permission) for the lib, it might not be best advertisement, when the demo app blocks with innocent looking samples. 

Maybe there should be at least a warning before critical tests hang the app?

### If it's a bug, version(s) of android this affects:

Android 9

### Version of osmdroid the issue relates to:

6.1.0 probably not only

**Comment by monsieurtanuki** [comment]
@InI4 I agree with you: the demo's goal is to show how osmdroid works (therefore it should work, not stall).
Could you be more specific about a demo that stalled, and about the needed permissions?

**Comment by InI4** [comment]
Me idiot. I just observed this while traveling and now I need to recheck. I do not remember, but it was obviously connected to the lack of storage access. Set this on wait, I need to reproduce first ...

**Comment by spyhunter99** [comment]
if you are rocking the debug variant built from source, it probably has one of the crash/heap analyzers running. It will freeze up when changing activities.
